### PR TITLE
Draft: Tweak test pipeline conan olaf

### PR DESCRIPTION
**Pull request by oschusler**

### CODE REVIEW

**Comment by pablolh** [comment]
I've identified two remaining problems:
- Linux ARM: since the architecture is arm, installation of Bison and Flex via Conan is skipped inside the container. Therefore, no Bison or Flex is present. CMake in tree-gen (which hasn't been updated to Conan) tries to build it and fail.

Suggested solution: install Bison and Flex manually with a workflow step in Linux ARM

- MacOS ARM: this is outside of any container, but the self-hosted runner has apparently Bison 2.3 which is too old as per tree-gen's CMake requirements. Building from source fails.
Suggested solution: either update the Bison version in the self-hosted runner to 3.0.0+, or lower the requirements in tree-gen's CMake.

**Comment by rturrado** [comment]
@Pablo, maybe we need to edit the Branch protection rules and remove the ARM64 builds from the Required list.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -3,12 +3,12 @@ name: Test
 on:
   push:
     branches:
-    - develop
+      - tweak-test-pipeline-conan
```
Comment (rturrado) [inline .github/workflows/test.yml:6]: Remember to keep this as `develop`.

**Comment by oschusler** [comment]
I don't really agree that ARM should be removed from the required list. These packages are public facing and should be tested on the systems they are supposed to support. If we remove them from the required list, this means that they are allowed to fail and we can continue without ARM support

File: setup.py [hunk 2]
```diff
@@ -102,141 +100,42 @@ def run(self):
...
-                        .split(".")
-                    ) >= (3, 12)
-                except:
-                    parallel_supported = False
-                if parallel_supported:
-                    cmd = cmd["--parallel"][nprocs]
-                elif not sys.platform.startswith("win"):
-                    cmd = cmd["--"]["-j"][nprocs]
+            cmd = (local['conan']['create']['.']
+                ['--version']['0.6.5']
```
Comment (rturrado) [inline setup.py:117]: We have to remember to update this version number.

This is something we will have to look into in the future. I don't like at all this hard coding of versions here and there. Maybe we could pass the version to `setup.py`, and, at the same time, automate that passing from GitHub, I don't really know. At the moment it is the way it is for QX simulator, libqasm...

**Review by rturrado — APPROVED** [review]
Thanks for the PR, Pablo!

**Comment by rturrado** [comment]
> I don't really agree that ARM should be removed from the required list. These packages are public facing and should be tested on the systems they are supposed to support. If we remove them from the required list, this means that they are allowed to fail and we can continue without ARM support

I don't agree either, Olaf. I was meaning: "if you want to merge this quickly, and not wait until we have the self-hosted runners properly set up". Other than that, I am more than 100% in line with your message.

File: setup.py (hunk 2 above)
Comment (pablolh) [reply setup.py]: I've replaced this version number with a call to get_version() which parses the Version.hpp header to have a single source of truth

File: .github/workflows/test.yml (hunk 1 above)
Comment (pablolh) [reply .github/workflows/test.yml]: Fixed thanks!

**Review by rturrado — APPROVED** [review]
Thanks Pablo! Good work!

File: setup.py [hunk 3]
```diff
@@ -114,7 +114,7 @@ def run(self):
             cmd & FG
 
             cmd = (local['conan']['create']['.']
-                ['--version']['0.6.5']
+                ['--version'][get_version()]
```
Comment (rturrado) [inline setup.py:117]: Beautiful, Pablo. I have to add this for `libqasm` too. Many thanks for this fix!

**Pull request merged**